### PR TITLE
Add design considerations

### LINKED ISSUE #219: Developer guide - Logic design considerations
Categorise the design considerations for logic subsection

### PR DESCRIPTION
**Pull request by Vangmay**
fixed #219

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -232,17 +232,18 @@ The activity diagram shows the general sequence of steps when a user interacts w
 
 ### Edit person feature
 
-#### Implementation
-
 #### Design considerations
+**Aspect: Check if the person with person id exists before editing**
```
Comment (rxchell) [inline docs/DeveloperGuide.md:236]: Minor change, but perhaps can use the code-formatted `personId` to be consistent with the code

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -263,18 +264,22 @@ The activity diagram shows the general sequence of steps when a user interacts w
 <br>
 
 ### Clear person feature
-
-#### Implementation
-
 #### Design considerations
+**Aspect: Should `clear persons` also delete appointments**
+- Alternative 1 (Current choice): The `clear persons` command should also clear the appointments else there will be alot of appointments inside the application that are linked to person ids that don't exists. Causing confusion for the users.
```
Comment (rxchell) [inline docs/DeveloperGuide.md:269]: Just some minor grammatical nitpicking and spelling typos, would it be possible to rephrase it? It would be good keep it consistent with the format under general commands as well :)

- **Alternative 1 (current choice):** The `clear persons` command should also clear all appointments.
  - Pros: This prevents the case where the appointments are linked to deleted `personIds` which do not exist. Hence, this prevents confusion for users.

Comment (rxchell) [inline docs/DeveloperGuide.md:269]: Should we specify that it clears **all** appointments?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -232,17 +232,18 @@ The activity diagram shows the general sequence of steps when a user interacts w
...
 <br>
 
 ### Delete person feature
 
-#### Implementation
-
 #### Design considerations
+**Aspect: deleting a person should also remove appointments linked to the person**
+- Alternative 1 (Current choice): Deleting a person will also remove any appointments with the personId of that person, this is to prevent any appointments with person ids that are non-existent.
+- Alternative 2: We assume the user deletes the appointment, however this is not a good assumption since the user might forget doing so.
```
Comment (rxchell) [inline docs/DeveloperGuide.md:246]: Would it be possible to rephrase it to keep it consistent with the formatting? Perhaps it can be rephrased as 

**Aspect: Deleting a person should also remove appointments linked to the person**
- **Alternative 1 (current choice):** Deleting a person will also remove any appointments with the `personId` of that person.
  - Pros: This prevents the case where appointments are linked to `personIds` that are non-existent.
- **Alternative 2:** Deleting a person will not remove any appointments with the `personId` of that person.
  - Cons: This assumes the user would delete the appointments linked to the deleted person's `personId`. However, the user might forget to do so.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -263,18 +264,22 @@ The activity diagram shows the general sequence of steps when a user interacts w
...
-
-#### Implementation
-
 #### Design considerations
+**Aspect: Should we implement as `addAppt` or `add appt`**
+- Alternative 1 (Current choice): Implement the add appointment feature as `add appt`
+  - Pros: Allows us to use the existing infrastructure, just have to add code to detect weather the entity is `appt` or not.
+  - Cons: Adds extra code to the file since more arguments need to be parsed, hence there is a chance of SLAP being violated.
+- Alternative 2: Implement the add appointment function as `addAppt`
+  - Pros: Creates a seperate command, so the implementations of `add person` and `add appointment` will be seperated from each other.
```
Comment (rxchell) [inline docs/DeveloperGuide.md:281]: Minor spelling error, should be "separate"

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -263,18 +264,22 @@ The activity diagram shows the general sequence of steps when a user interacts w
...
 <br>
 
 ### Add appointment feature
-
-#### Implementation
-
 #### Design considerations
+**Aspect: Should we implement as `addAppt` or `add appt`**
+- Alternative 1 (Current choice): Implement the add appointment feature as `add appt`
+  - Pros: Allows us to use the existing infrastructure, just have to add code to detect weather the entity is `appt` or not.
```
Comment (rxchell) [inline docs/DeveloperGuide.md:278]: Minor spelling error, should be "whether"

**Review by rxchell — CHANGES_REQUESTED** [review]
LGTM, some suggestions!

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (rxchell) [reply docs/DeveloperGuide.md:278]: Do you think this aspect can be generalised for the other commands as well, since they also check for the entity type?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -232,17 +232,18 @@ The activity diagram shows the general sequence of steps when a user interacts w
 
 ### Edit person feature
 
-#### Implementation
-
 #### Design considerations
+**Aspect: Check if the person with person id exists before editing**
+- This is to ensure no unwanted errors occur while editing the person.
```
Comment (rxchell) [inline docs/DeveloperGuide.md:237]: Perhaps we can add that this helps maintain data integrity.

**Comment by jayjay19630** [comment]
Once this is merged and the formatting is figured out, I'll add on to this to complete the logic design considerations

**Review by rxchell — APPROVED** [review]
LGTM, thanks for making the changes!
